incubator/hnc/internal/validators/hierarchy_test.go

```diff
@@ -96,6 +98,54 @@ func TestChangeParentOnManagedBy(t *testing.T) {
 	}
 }
 
+func TestChangeParentWithConflict(t *testing.T) {
+	f := foresttest.Create("-a-c") // a <- b; c <- d
+
+	// Set secret to "Propagate" mode.
+	or := &reconcilers.ObjectReconciler{
+		GVK:  schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Secret"},
+		Mode: api.Propagate,
+	}
+	f.AddTypeSyncer(or)
```

Comment:
Why not just use rolebindings? Seems easier, but if there's a reason, comment it

Reply:
The test forest doesn't have `Role` and `RoleBinding` as `TypeSyncer` by default either. So it's the same for RB and secret, and we have an existing `createSecret()` func. Will comment that.

Reply:
sg